### ISSUE TITLE
Dedicated buttons and shortcuts for creating new global or local variables

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6940]
### Why this would be helpful

Depending on the project, you might use a lot of local variables. If there was a faster way to set newly created variables to "For this sprite only", I bet those creators would really like it.

### How this addon works

My idea to make this faster is to:
1. Replace the radio buttons for choosing the variable's scope with two buttons, both for creating the variable, but one of them would make it global, and the other one would make it local.
2. Assign keyboard shortcuts to these buttons.

![New Variable modal with dedicated buttons to create scoped variables with visible keyboard shortcuts](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/48f19c8e-2c3b-41d1-84f7-53a3e82b29c2)

### Possible alternatives

I know there are also a bunch of other ideas.
- Just use a keyboard shortcut for creating a new local variable instead of adding dedicated buttons
- Scoping it based on the variable's name: #6594
- Defaulting to making new local variables: #6592
- Remembering if you had selected "For all sprites" or "For this sprite only" and apply that to the next variable created within a short amount of time
- A better menu: #6742

**Comment by WorldLanguages** [comment]
This looks superior and more intuitive than other suggestions. It could be extended to allow for creating multiple variables of the same type, for example by showing an additional input when pressing Tab.

**Comment by WorldLanguages** [comment]
Also, your concept screenshot forgets about the existence of cloud variables, I think.

**Comment by DNin01** [comment]
> Also, your concept screenshot forgets about the existence of cloud variables, I think.

Oh, yeah. I was on scratchfoundation.github.io, so the checkbox didn't show up. But still, for Cloud variables, we could simply gray out the button for creating a local variable.

**Comment by WorldLanguages** [comment]
Yeah, I would also avoid thinking about creating multiple cloud variables at once. Users can create them one at a time.

**Comment by Jazza-231** [comment]
> This looks superior and more intuitive than other suggestions.

Yes I do actually agree, should I instead go down this route? It does look a lot nicer and does make more sense UI wise

**Comment by WorldLanguages** [comment]
@DNin01 You could consider extending your concept screenshot. We could add a small sentence that clarifies how to create more than one variable at once. For example, `Press Tab to create multiple variables`.

**Comment by Jazza-231** [comment]
My current plan is to modify this UI to also include initial value, change the shift to ctrl, and obviously include cloud variables. I think this could make variable creation a lot less painful, are people generally happy with this idea? I'm not keen on creating multiple variables because of the complexity, and also it would mess up keybinds for quickly making variables of a certain type. I will provide a mock-up in the near future.

**Comment by Samq64** [comment]
I think WorldLanguages meant something similar to the GitHub topics selector but using tab insed of space to move to the next one and the keybinds/buttons would make all of those local or global. The first version of the addon doesn't have to have multiple variable support though.
![image](https://github.com/ScratchAddons/ScratchAddons/assets/81489795/8aeba097-8939-4e63-94b3-17cf4224bbbd)

**Comment by Secret-chest** [comment]
Cloud variables shouldn't have a key.

**Comment by WorldLanguages** [comment]
> Cloud variables shouldn't have a key.

We will pretty much ignore cloud variables, as they're not created very frequently.

**Comment by BroJac5246** [comment]
> > Cloud variables shouldn't have a key.
> 
> We will pretty much ignore cloud variables, as they're not created very frequently.

Why not? It's not that hard to press Ctrl + Shift + Enter or Alt + Enter, and it's probably not that hard to code it, either. We don't have to display it prominently, but it should at least be an option.
